### PR TITLE
Skip allocations when reusing VP9 packets

### PR DESCRIPTION
**Pull request by kcaffrey**
If a caller reuses `codecs.VP9Packet` (after resetting the state appropriately), then reuse already allocated slices if they are of sufficient length instead of unconditionally allocating. Add tests to verify that reuse is safe and is zero allocations.

### CODE REVIEW

File: codecs/vp9_packet.go [hunk 1]
```diff
@@ -451,9 +451,17 @@ func (p *VP9Packet) parseSSData(packet []byte, pos int) (int, error) { // nolint
...
+		if cap(p.Width) < int(NS) {
+			p.Width = make([]uint16, NS)
+		} else {
+			p.Width = p.Width[:NS]
+		}
+		if cap(p.Height) < int(NS) {
+			p.Height = make([]uint16, NS)
+		} else {
+			p.Height = p.Height[:NS]
+		}
```
Comment (JoTurk) [inline codecs/vp9_packet.go]: Nit: instead of adding a nolint i suggest you refactor this into a small helper resizeUint16Slice or something.

File: codecs/vp9_packet.go [hunk 2]
```diff
@@ -482,10 +490,15 @@ func (p *VP9Packet) parseSSData(packet []byte, pos int) (int, error) { // nolint
 
 		p.PGTID = append(p.PGTID, packet[pos]>>5)
 		p.PGU = append(p.PGU, packet[pos]&0x10 != 0)
-		R := (packet[pos] >> 2) & 0x3
+		R := (packet[pos] >> 2) & 0x3 // nolint: varnamelen
```
Comment (JoTurk) [inline codecs/vp9_packet.go]: Can you rename this to reference or something instead of adding a nolint.

**Review by JoTurk — COMMENTED** [review]
Change looks good, thank you.
just two nits.

File: codecs/vp9_packet.go (hunk 1 above)
Comment (kcaffrey) [reply codecs/vp9_packet.go]: Good call, done

File: codecs/vp9_packet.go (hunk 2 above)
Comment (kcaffrey) [reply codecs/vp9_packet.go]: Done

**Review by JoTurk — APPROVED** [review]
thank you.

**Comment by kcaffrey** [comment]
Thanks for the review @JoTurk. Any chance you can create a new release with this? I saw the workflow for releasing but I don't have access to run it.

**Comment by JoTurk** [comment]
@kcaffrey you do have access. Just tag v1.10.3 off main and push it and it will create the release :)
